### PR TITLE
created db table for collegiatesubreddit and controller

### LINKED ISSUE #5: Create `CollegiateSubredditController`, add `GET` (index) and `POST` (create)
Create `CollegiateSubredditController`, add `GET` (index) and `POST` (create)

# Acceptance Criteria:

- [x] There is a controller file `CollegiateSubredditController.java`
      in the expected directory.
- [x] In `CollegiateSubredditController.java` there is 
      code for a `GET /api/collegiateSubreddits/all` endpoint 
      that returns a JSON list of all subreddits in the database.
      (We sometimes call this an *index* action since it lists all
      items in the database.)
- [x] In `CollegiateSubredditController.java` there is 
      code for a `POST /api/collegiateSubreddits/post` endpoint
      that can be used to create a new entry in the table. (This
      is a *create* action.)
- [x] The Swagger-UI endpoints for these are well documented so that
      any member of the team can understand what they are for and
      how to use them.
- [x] The `POST` endpoint works as expected, in the sense that new
      records can be added to the database (on localhost).
- [x] The `GET` endpoint works as expected, in the sense that the new
      records that are added show up (on localhost).
- [x] The `GET` and `POST` endpoints work as expected when the 
      app is deployed to Heroku.
- [x] There is full test coverage (Jacoco) for the methods in 
      `CollegiateSubredditController.java`
- [x] There is full mutation test coverage (Pitest) for the methods in
      `CollegiateSubredditController.java`

# Details

Consult the methods in [`TodosController.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/main/java/edu/ucsb/cs156/example/controllers/TodosController.java) for examples 
of controller methods.

Consult the tests in [`TodosControllerTests.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/test/java/edu/ucsb/cs156/example/controllers/TodosControllerTests.java) for examples
of controller tests.

### PR DESCRIPTION
**Pull request by DannyCoco**
created db table for collegiatesubreddit and controller, get(index), post(create)

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "List all collegiate subreddits in database")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<CollegiateSubreddit> getCollegiateSubreddits() {
+        loggingService.logMethod();
+        Iterable<CollegiateSubreddit> collegiateSubs = collegiateSubredditRepository.findAll();
+        return collegiateSubs;
+    }
```
Comment (jacksoncooper) [inline src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java:49]: I think this endpoint should not be accessible to `USERS`. I was reading [a thread](https://ucsb-cs156-w22.slack.com/archives/C030KT692A3/p1643670228956679?thread_ts=1643668881.909719&cid=C030KT692A3) by Prof. Conrad last night about this on Slack.

**Review by jacksoncooper — DISMISSED** [review]
Please see annotations.

File: src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java (hunk 1 above)
Comment (jacksoncooper) [reply src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java:49]: Oh, I'm wrong. Dismissing.
> For our three database tables, though, all records are available to everyone.

**Comment by sytarno** [comment]
We checked it on heroku for the final pass and it looks good!

**Pull request merged**